docs/source/tutorials/python_in_mantid/solutions/02_pim_sol.rst

```diff
@@ -0,0 +1,141 @@
+
+.. plot::
+
+    # import mantid algorithms, numpy and matplotlib
+    from mantid.simpleapi import *
+    import matplotlib.pyplot as plt
+    import numpy as np
+
+    _164198 = Load('164198')
+
+    fig, axes = plt.subplots(edgecolor='#ffffff', num='164198-1', subplot_kw={'projection': 'mantid'})
+    axes.plot(_164198, color='#2ca02c', label='164198: spec 100', linewidth=1.0, specNum=100, zorder=2.1)
+    axes.plot(_164198, color='#1f77b4', label='164198: spec 200', linewidth=1.0, specNum=200, zorder=2.1)
+    axes.errorbar(_164198, capsize=1.0, color='#ff7f0e', label='A funky label', linewidth=1.0, specNum=50)
+    axes.plot(_164198, color='#000000', label='164198: spec 300', linewidth=1.0, markeredgecolor='#d62728', markerfacecolor='#d62728', specNum=300, zorder=2.1)
+    axes.set_title('My Beautiful Plot')
+    axes.set_xlabel('Time-of-flight ($\mu s$)')
+    axes.set_ylabel('Counts')
+    axes.set_xlim([460.0, 600.0])
+    axes.set_ylim([1.0, 2000.0])
+    axes.set_yscale('log')
+    axes.legend() #.set_draggable(True)
+
+    #plt.show()
+   
+
+C - Processing and Plotting SNS Data
+====================================
+
+.. plot::
+   :include-source:
+
+    # import mantid algorithms, numpy and matplotlib
+    from mantid.simpleapi import *
+    import matplotlib.pyplot as plt
+    import numpy as np
+
+    Load(Filename=r'EQSANS_6071_event.nxs',OutputWorkspace='run',LoadMonitors='1')
+    ConvertUnits(InputWorkspace='run_monitors',OutputWorkspace='run_monitors_lambda',Target='Wavelength')
+    Rebin(InputWorkspace='run_monitors_lambda',OutputWorkspace='run_monitors_lambda_rebinned',Params='2.5,0.1,5.5')
+    ConvertUnits(InputWorkspace='run',OutputWorkspace='run_lambda',Target='Wavelength')
+    Rebin(InputWorkspace='run_lambda',OutputWorkspace='run_lambda_rebinned',Params='2.5,0.1,5.5')
+    SumSpectra(InputWorkspace='run_lambda_rebinned', OutputWorkspace='run_lambda_summed')
+    Divide(LHSWorkspace='run_lambda_summed', RHSWorkspace='run_monitors_lambda_rebinned', OutputWorkspace='normalized')
+
+    from mantid.api import AnalysisDataService as ADS
+
+    run_lambda_summed = ADS.retrieve('run_lambda_summed')
+    run_monitors_lambda_rebinned = ADS.retrieve('run_monitors_lambda_rebinned')
+    normalized = ADS.retrieve('normalized')
+
+    fig, axes = plt.subplots(edgecolor='#ffffff', num='run_lambda_summed-1', subplot_kw={'projection': 'mantid'})
+    axes.plot(run_lambda_summed, color='#1f77b4', label='run_lambda_summed: spec 1', linewidth=1.0, specNum=1)
+    axes.plot(run_monitors_lambda_rebinned, color='#ff7f0e', label='run_monitors_lambda_rebinned: spec 1', linewidth=1.0, specNum=1)
+    axes.plot(normalized, color='#2ca02c', distribution=False, label='normalized: spec 1', linewidth=1.0, specNum=1)
+    axes.set_title('run_lambda_summed')
+    axes.set_xlabel('Wavelength ($\AA$)')
+    axes.set_ylabel('($\AA$)$^{-1}$')
+    axes.set_xlim([2.405, 4.5])
+    axes.set_yscale('log')
+    axes.legend() #.set_draggable(True) 
```

Comment:
I feel when people are reading this solution they'll possibly wonder why this comment is here. But I have no issues with leaving it in if they are supposed to use it.
```suggestion
    axes.legend()
```

Reply:
Again this is because the solution comes from a generated script. But `.set_draggable()` is from a newer mpl version than our build servers, so I had to comment it out. I left the comment here so that the solution matches the script generation that the user will produce.